src/classes/HH_HouseholdNaming.cls

```diff
@@ -210,236 +181,149 @@ public without sharing class HH_HouseholdNaming {
-                listCon.add(con);
+    public Set<String> getHouseholdNamingContactFields() {
+        if (!settings.isAdvancedHouseholdNaming()
+                || namingService.setHouseholdNameFieldsOnContact() == null) {
+            return new Set<String>();
+        } else {
+            return namingService.setHouseholdNameFieldsOnContact();
         }
-        return iNaming.getHouseholdFormalGreeting(listCon);
     }
 
     /*******************************************************************************************************
-    * @description Returns Contact fields specified in the Household Naming Settings format fields.
-    * @return Set<String> If the Automatic Household Naming is enabled, return set of Contact field API names; 
-    * otherwise, an empty set
+    * @description the class object that supports the HH_INaming interface
     */
-    public Set<String> getHouseholdNamingContactFields() {
-
-        if (!isAutomaticNaming()) {
-            return new Set<String>();
+    HH_INaming namingService {
+        get {
+            if (namingService == null) {
+                String implementingClass = new HouseholdSettings().getImplementingClass();
+
+                Type classType = Type.forName(implementingClass);
+                if (classType != null) {
+                    Object classInstance = classType.newInstance();
+                    if (classInstance instanceof HH_INaming) {
+                        namingService = (HH_INaming) classInstance;
+                    }
+                }
+            }
+            return namingService;
         }
-    
-        Set<String> fieldNames = iNaming.setHouseholdNameFieldsOnContact();
-        return fieldNames == null ? new Set<String>() : fieldNames;
+        set;
     }
 
     /*******************************************************************************************************
     * @description executes the batch job to update all household names
     * @param isActivation whether this is being called when npo02__Advanced_Household_Naming__c is being turned on
     * @return void
     */
-    public static void refreshAllHouseholdNaming(boolean isActivation) {
+    public static void refreshAllHouseholdNaming(Boolean isActivation) {
         // the household batch expects a list of Contact's with just Id, LastName, HHId__c available.
-        string strSoql = 'SELECT Id, LastName, ' + UTIL_Namespace.StrTokenNSPrefix('HHId__c') + ' FROM Contact WHERE ' +
-            UTIL_Namespace.StrTokenNSPrefix('HHId__c') + ' != NULL ';
+        String strSoql = 'SELECT Id, LastName, ' + UTIL_Namespace.StrTokenNSPrefix('HHId__c') + ' FROM Contact WHERE ' +
+                UTIL_Namespace.StrTokenNSPrefix('HHId__c') + ' != NULL ';
         HH_HouseholdNaming_BATCH batch = new HH_HouseholdNaming_BATCH(strSoql, isActivation);
-        integer batchSize = 200;
-        id batchProcessId = database.executeBatch(batch, batchSize);
+        Integer batchSize = 200;
```

Comment:
let's use a constant for 200. There may be one declared already.

Reply:
Actually, I think this should be removed since the default batch size is 200, but I don't really want to change this within this work stream.  I'm going to leave it as is for now with a goal of factoring this out later.

Reply:
It's worth defining as a constant or getter in the `HH_HouseholdNaming_BATCH` class. That'll make it easier for us to eventually implement a setting for this.

Reply:
Extracted it into a method!